xbmc/video/VideoInfoScanner.cpp

```diff
@@ -1617,9 +1661,59 @@ namespace VIDEO
       }
       else
       {
-        CLog::Log(LOGDEBUG,"%s - no match for show: '%s', season: %d, episode: %d.%d, airdate: '%s', title: '%s'",
-                  __FUNCTION__, showInfo.m_strTitle.c_str(), file->iSeason, file->iEpisode, file->iSubepisode,
-                  file->cDate.GetAsLocalizedDate().c_str(), file->strTitle.c_str());
+        if (!CSettings::GetInstance().GetBool(CSettings::SETTING_VIDEOLIBRARY_IMPORTALL))
+        {
+          CLog::Log(LOGDEBUG,"%s - no match for show: '%s', season: %d, episode: %d.%d, airdate: '%s', title: '%s'",
+                    __FUNCTION__, showInfo.m_strTitle.c_str(), file->iSeason, file->iEpisode, file->iSubepisode,
+                    file->cDate.GetAsLocalizedDate().c_str(), file->strTitle.c_str());
+        }
+        else
+        {
+          CFileItem item;
+          item.SetPath(file->strPath);
+          item.GetVideoInfoTag()->m_strTitle = CURL(file->strPath).GetFileNameWithoutPath();
+          CURL::Decode(item.GetVideoInfoTag()->m_strTitle);
+          item.GetVideoInfoTag()->m_strSortTitle = item.GetVideoInfoTag()->m_strTitle;
+          item.GetVideoInfoTag()->m_iSeason = file->iSeason;
+          item.GetVideoInfoTag()->m_iEpisode = file->iEpisode;
+
+          // Let's try to get the modification datetime
+          CDateTime dateAdded;
+          struct __stat64 buffer;
+          if (CFile::Stat(file->strPath, &buffer) == 0 && (buffer.st_mtime != 0 || buffer.st_ctime !=0))
+          {
+            time_t now = time(NULL);
+            time_t addedTime;
+            // Prefer the modification time if it's valid
+            if (g_advancedSettings.m_iVideoLibraryDateAdded == 1)
+            {
+              if (buffer.st_mtime != 0 && (time_t)buffer.st_mtime <= now)
+                addedTime = (time_t)buffer.st_mtime;
+              else
+                addedTime = (time_t)buffer.st_ctime;
+            }
+            // Use the newer of the creation and modification time
+            else
+            {
+              addedTime = std::max((time_t)buffer.st_ctime, (time_t)buffer.st_mtime);
+              // if the newer of the two dates is in the future, we try it with the older one
+              if (addedTime > now)
+                addedTime = std::min((time_t)buffer.st_ctime, (time_t)buffer.st_mtime);
+            }
+
+            // make sure the datetime does is not in the future
+            if (addedTime <= now)
+            {
+              struct tm *time = localtime(&addedTime);
+              if (time)
+                dateAdded = *time;
+            }
+          }
+
+          item.GetVideoInfoTag()->m_firstAired = dateAdded;
```

Comment:
I'm pretty sure this whole logic (deducing added time from file times) exists somewhere else too (can't remember where). It's how `date added` works for library items.